### PR TITLE
feat: add metadata check script

### PR DESCRIPTION
**Pull request by Baoyuantop**
This script checks the following rules for Markdown files in the given directories:
1. File name should only contain [0-9, a-z, -, .]. It checks this with the regex fileNameRegex.
2. Each file should have metadata in YAML frontmatter. It checks this by parsing the frontmatter with gray-matter and checking that Object.keys(metadata).length > 0.
3. For files in directories where "slug" is in customCheckList, the slug should:
- Only contain [0-9, a-z, -, .]
- Start with /
4. It checks all files in the given directories, and prints any failures. If all files pass, it prints a success message for that directory.
5. If any file fails a check, isPassed is set to false. At the end of the script, isPassed is checked, and if it's false the script exits with exit code 1, indicating failure.

### CODE REVIEW

**Comment by kayx23** [comment]
Just wanted to add some background info as to why a whitespace shouldn't be allowed in this project:
* Docusaurus does handle whitespace in markdown frontmatter slug pretty well, by HTML encoding the space and automatically updating the left navbar ([discussed here](https://github.com/facebook/docusaurus/issues/8843#issuecomment-1499785461)) 
* however, as our projects use custom components like navbar, which are not automatically generated or designed to handle whitespace in slug situation, we should enforce this rule to not allow whitespace, to avoid downstream headaches
* Also I personally don't see any situation where a whitespace in URL is preferred in our use case

File: scripts/meta-check.js [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+dirList.forEach((item) => {
+  const dir = item.path;
+  const customCheckList = item.customCheckList;
+
+  const allFiles = getAllFiles(dir);
+
+  allFiles.forEach((file, index) => {
+    if (file.endsWith('.md') && path.parse(file).name !== "README") {
+
+      // 检查文件名称
```
Comment (kayx23) [inline scripts/meta-check.js:38]: Do you mind listing what this script checks in the PR description? Easier for others to quickly skim through that way. Thanks!!

File: scripts/meta-check.js [hunk 2]
```diff
@@ -0,0 +1,75 @@
+const matter = require('gray-matter');
+const fs = require('fs');
+const path = require('path');
+const process = require('process');
+
+const dirList = [
+  { path: '../apisix_versioned_docs', customCheckList: ['title', 'slug'] },
```
Comment (juzhiyuan) [inline scripts/meta-check.js:7]: Except for APISIX & Enterprise, we also have other products documents. Please scan and check all Markdown Files, except for the items in whitelist

Comment (guoqqqi) [reply scripts/meta-check.js:7]: As we are mainly updating the documentation for EE and APISIX, the documentation rules for the rest of the products can be updated later

**Review by guoqqqi — APPROVED** [review]

**Pull request merged**